### PR TITLE
Reworked how timestamps are calculated when you post videos.

### LINKED ISSUE #43: Wrong timestamp on videos
Timestamp on videos is wrong for videos who is not 30min and using the serial number to calculate it.

Example:
timestamp-bug

Reproduce by adding two videos not 30 min. long with same timestamp in filename and incremented serial number.

### PR DESCRIPTION
**Pull request by R0flcopt3r**
Timestamp for the next video is now calculated based on the length of the one before it. When adding a new video that has an offset, it's based on the length of that particular video. It is also possible to add videos with different timestamps in the file name, and it should figure out the correct timestamp for it and all the ones after.

Still missing tests, figured I could get some feedback on this before spending too much time with that.

The creation of the video isn't ideal. I temporary create a timestamp that I later overwrite. Not sure if I can solve this more elegantly without refactoring Video to have an optional timestamp. That would also require updating the ORM, so I would very much like to avoid that.

Every video is also parsing its filename to find the timestamp, which is then later overwritten in the API. This is handy, as I don't have to think about it not having a potentially real timestamp, and can easily just use it when calculating the next timestamp. This could also probably be handled more elegantly.

closes #43

### CODE REVIEW

File: src/core/model.py [hunk 1]
```diff
@@ -507,9 +507,7 @@ def is_processed(self) -> bool:
         return True
 
 
-def parse_str_to_date(
-    string: str, offset_time: int = 1800
-) -> Optional[datetime]:
+def parse_str_to_date(string: str, offset_time: int = 1800) -> datetime | None:
```
Comment (R0flcopt3r) [inline src/core/model.py:510]: why is pre-commit changing to this convention? isn't this Python 3.10?

Comment (MindTooth) [reply src/core/model.py:510]: Handled by the future import at the top.

https://github.com/asottile/pyupgrade does the rewrite.

**Comment by tomrtk** [comment]
I have not run it, but seems ok for me. I guess we could find a more elegant solution, but if this works, go for it for now.

I guess we should release `v1.0.0` after this is merged.

**Review by tomrtk — APPROVED** [review]
Do you plan to update the tests?

**Comment by R0flcopt3r** [comment]
Great! I'll look into the tests tomorrow

**Comment by R0flcopt3r** [comment]
Obviously I didn't end up doing this. Had some computer issues, and were unable to boot my Linux environment in order to work on this.

**Comment by R0flcopt3r** [comment]
@tomrtk could you look over once more, now that the tests are in place.

**Comment by R0flcopt3r** [comment]
My tests aren't complete, and I think I have some zombie code. I will fix that before merge.

File: src/core/api/api.py [hunk 2]
```diff
@@ -306,9 +308,33 @@ def add_job_to_project(
         errors: Dict[str, List[str]] = dict()
         file_not_found = []
         time_not_found = []
+        next_timestamp: Optional[datetime] = None
```
Comment (tomrtk) [inline src/core/api/api.py:311]: If you feel like it, most of this function could benefit from a refactor to a `service` function for the job and video creation.

Comment (tomrtk) [reply src/core/api/api.py:311]: Would probably also make testing it easier.

**Review by tomrtk — COMMENTED** [review]
Else it looks ok for me.

File: src/core/api/api.py (hunk 2 above)
Comment (R0flcopt3r) [reply src/core/api/api.py:311]: yeah, that is a good idea. I think moving the logic out would be the final step for this PR. It currently works and passes all the test cases I could come up with.

**Pull request closed without merging**